lib/auth/user.go

```diff
@@ -24,38 +24,48 @@ limitations under the License.
 package auth
 
 import (
-	"github.com/gravitational/teleport"
+	"context"
+
 	"github.com/gravitational/teleport/lib/events"
 	"github.com/gravitational/teleport/lib/services"
 
 	"github.com/gravitational/trace"
 )
 
-// UpsertUser upserts user
+// CreateUser inserts a new entry in a backend.
+func (s *AuthServer) CreateUser(ctx context.Context, user services.User) error {
+	err := s.Identity.CreateUser(user)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	s.EmitAuditEvent(events.UserCreate, events.EventFields{
+		events.EventUser:       user.GetCreatedBy().User.Name,
+		events.UserExpires:     user.Expiry(),
+		events.UserRoles:       user.GetRoles(),
+		events.UserCreatedName: user.GetName(),
+	})
+
+	return nil
+}
```

Comment:
@benarent This event will only be issued for local users. We probably want a analogous event for SSO users right? Or do you think a "certificate was issued" type of event would be more appropriate for SSO users?

Reply:
I think it's interesting addition but for now I would keep this just to local users.  As we get feedback from customers on how they want to manage audit events from the identity provider. I can see a 'first time' login being potentially useful but I don't want to complicate this PR.

Reply:
We should definitely raise an issue to make sure we implement this in future. I can certainly imagine SSO users being very interested in knowing about the first time a user connected to a cluster and had an identity provisioned - we should have feature parity between local users and SSO users here.